java/src/org/openqa/selenium/grid/node/relay/RelaySessionFactory.java

```diff
@@ -149,15 +147,6 @@ public Either<WebDriverException, ActiveSession> apply(CreateSessionRequest sess
               "New session request capabilities do not " + "match the stereotype."));
     }
 
-    // remove browserName capability if 'appium:app' is present as it breaks appium tests when app
```

Comment:
The `merge` operation that necessitated this special-case handling has been eliminated. The session factor shouldn't be altering clients' requested capabilities.